### ISSUE TITLE
Not working on Python 3.9.2

### ISSUE THREAD

**Issue by kjohnsen** [brian-team/brian2#1284]
I'm on Windows 10, and the Python version that got installed was 3.9.2, as stated in the title. I took these steps:
```bash
conda create -n brian2_py39 python=3.9
conda activate brian2_py39
conda install brian2
python
```

Then once in Python:
```import brian2```

I get the following error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\__init__.py", line 66, in <module>
    from brian2.only import *
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\only.py", line 24, in <module>
    from brian2.core.clocks import *
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\core\clocks.py", line 13, in <module>
    from brian2.groups.group import VariableOwner
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\groups\__init__.py", line 5, in <module>
    from .group import *
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\groups\group.py", line 28, in <module>
    from brian2.codegen.codeobject import create_runner_codeobj
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\codegen\__init__.py", line 6, in <module>
    from .runtime import *
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\codegen\runtime\__init__.py", line 16, in <module>
    from .numpy_rt import *
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\codegen\runtime\numpy_rt\__init__.py", line 9, in <module>
    from .numpy_rt import *
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\codegen\runtime\numpy_rt\numpy_rt.py", line 19, in <module>
    from ...generators.numpy_generator import NumpyCodeGenerator
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\codegen\generators\__init__.py", line 10, in <module>
    from .cpp_generator import *
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\codegen\generators\cpp_generator.py", line 6, in <module>
    from brian2.codegen.cpp_prefs import C99Check
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\site-packages\brian2\codegen\cpp_prefs.py", line 46, in <module>
    previously_stored_flags = json.load(f, encoding='utf-8')
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\json\__init__.py", line 293, in load
    return loads(fp.read(),
  File "C:\ProgramData\Anaconda3\envs\brian2_py39\lib\json\__init__.py", line 359, in loads
    return cls(**kw).decode(s)
TypeError: __init__() got an unexpected keyword argument 'encoding'
```

**Comment by mstimberg** [comment]
Hi, thanks for reporting the problem – we should definitely enable Python 3.9 in our test suite... I can't test it right now, but I think we could simply delete the `encoding` argument here, or move it into the `open(...)` call:
https://github.com/brian-team/brian2/blob/b750fea045fc23bbf5e795ba6e7930dcc5be8747/brian2/codegen/cpp_prefs.py#L47-L48
Would be great if you could verify whether this works.

**Comment by kjohnsen** [comment]
I tried moving it to the open call and it imports. I'll make a pull request so we can see if it passes the tests.

**Comment by mstimberg** [comment]
Fixed via #1286 .
